pkg/didcomm/protocol/didexchange/service.go

```diff
@@ -296,14 +285,44 @@ func (s *Service) sendMsgEvents(msg *service.StateMsg) {
 func (s *Service) startInternalListener() {
 	go func() {
 		for msg := range s.callbackChannel {
-			// TODO handle error in callback - https://github.com/hyperledger/aries-framework-go/issues/242
-			if err := s.process(msg); err != nil {
-				// TODO handle error
-				logger.Errorf(err.Error())
+			if msg.Err != nil {
+				// TODO https://github.com/hyperledger/aries-framework-go/issues/438 Cleanup/Update data on Stop Event Action
+				logger.Errorf("client action event processing failed - msgID:%s error:%s", msg.ID, msg.Err)
+				s.processCallbackFailure(msg.ID, msg.Err)
```

Comment:
Are these really "callback" failures?

Reply:
At the moment, only callback failures are handled. But the method support any protocol failure. Will update this.

Reply:
[Updated](https://github.com/hyperledger/aries-framework-go/pull/480/files#diff-f761c91e753eba1c42a68076df06ce12R334).